lib/src/git_backend.rs

```diff
@@ -376,8 +367,20 @@ impl Backend for GitBackend {
         }
 
         let locked_repo = self.repo.lock().unwrap();
-        let git_commit_id = Oid::from_bytes(id.as_bytes())?;
-        let commit = locked_repo.find_commit(git_commit_id)?;
+        let git_commit_id =
+            Oid::from_bytes(id.as_bytes()).map_err(|err| BackendError::InvalidHash {
+                hash: id.hex(),
+                object_type: "commit",
+                source: Box::new(err),
+            })?;
+        let commit =
+            locked_repo
+                .find_commit(git_commit_id)
+                .map_err(|err| BackendError::ObjectNotFound {
+                    hash: id.hex(),
+                    object_type: "commit",
+                    source: Box::new(err),
+                })?;
```

Comment:
I think we still want to handle `git2::ErrorCode::NotFound` specially here. I haven't checked, but I suspect we could get other errors if a packfile was corrupt, for example. In such cases, it seems better to return that error than to say that the object doesn't exist. Practically, it would mean that the user gets informed about the corrupt repo instead of being told that the object commit doesn't exist. Maybe that'll help them a bit?